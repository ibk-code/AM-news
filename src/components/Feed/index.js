import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class Feed extends React.Component {
  /**
   * image errror function
   * @param {object} e
   */
  imgError = (e) => {
    e.target.src = './asset/img/dummy.jpg';
  };

  render() {
    const article = this.props.article;
    return (
      <React.Fragment>
        <div className="feed">
          <Link to={`/detail?title=${article.abstract}`}>
            <div className="feed__img">
              <img
                src={article?.img || './asset/img/dummy.jpg'}
                onError={this.imgError}
                alt={article.abstract}
                className="img-fluid"
              />
              <div className="gradient"></div>
              <p className="date">
                Date: {new Date(article.pub_date).toDateString()}
              </p>
            </div>
            <div className="desc">
              <p className="author">
                Author:{' '}
                {article.source === null ? 'Not available' : article.source}
              </p>
              <p className="title">{article.headline.main}</p>
            </div>
          </Link>
        </div>
      </React.Fragment>
    );
  }
}

Feed.propTypes = {
  url: PropTypes.string,
  image: PropTypes.string,
  title: PropTypes.string,
  author: PropTypes.string,
  source: PropTypes.string,
  published_at: PropTypes.string
};

export default Feed;
