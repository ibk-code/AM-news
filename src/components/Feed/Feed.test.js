import { shallow } from 'enzyme';
import React from 'react';
import Feed from './index';

describe('Feed Component test', () => {
  it('expect to render feed component', () => {
    const article = {
      title: 'Test Article',
      image:
        'https://cdn.cnn.com/cnnnext/dam/assets/210112041400-02-ireland-covid-0106-super-169.jpg',
      url: 'http://rss.cnn.com/~r/rss/edition_europe/~3/3a_KseS2B9A/index.html',
      author: 'ibk-code',
      source: 'cnn',
      published_at: '2021-01-13T07:10:15+00:00'
    };

    expect(shallow(<Feed article={article} />)).toMatchSnapshot();
  });
});
