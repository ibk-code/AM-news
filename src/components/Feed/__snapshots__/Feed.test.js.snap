// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed Component test expect to render feed component 1`] = `
<Fragment>
  <div
    className="feed"
  >
    <Link
      to="/detail?title=Test Article"
    >
      <div
        className="feed__img"
      >
        <img
          alt="Test Article"
          className="img-fluid"
          onError={[Function]}
          src="https://cdn.cnn.com/cnnnext/dam/assets/210112041400-02-ireland-covid-0106-super-169.jpg"
        />
        <div
          className="gradient"
        />
        <p
          className="date"
        >
          Date: 
          Wed Jan 13 2021
        </p>
      </div>
      <div
        className="desc"
      >
        <p
          className="author"
        >
          Author:
           
          cnn
        </p>
        <p
          className="title"
        >
          Test Article
        </p>
      </div>
    </Link>
  </div>
</Fragment>
`;
